ParseLoginUI/src/main/java/com/parse/ParseQueryRecyclerViewAdapter.java

```diff
@@ -0,0 +1,611 @@
+
+                notifyOnLoadedListeners(foundObjects, e);
+            }
+        });
+    }
+
+    /**
+     * Loads the next page of objects, appends to table, and notifies the UI that the model has
+     * changed.
+     */
+    public void loadNextPage() {
+        if (objects.size() == 0 && runningQueries.size() == 0) {
+            loadObjects(0, false);
+        }
+        else {
+            loadObjects(currentPage + 1, false);
+        }
+    }
+
+    private View getDefaultView() {
+        if (this.itemResourceId != null) {
+            return View.inflate(context, itemResourceId, null);
+        }
+        LinearLayout view = new LinearLayout(context);
+        view.setPadding(8, 4, 8, 4);
+
+        ParseImageView imageView = new ParseImageView(context);
+        imageView.setId(android.R.id.icon);
+        imageView.setLayoutParams(new LinearLayout.LayoutParams(50, 50));
+        view.addView(imageView);
+
+        TextView textView = new TextView(context);
+        textView.setId(android.R.id.text1);
+        textView.setLayoutParams(new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,
+                ViewGroup.LayoutParams.WRAP_CONTENT));
+        textView.setPadding(8, 0, 0, 0);
+        view.addView(textView);
+
+        return view;
+    }
+
+    @Override
+    public ViewHolder onCreateViewHolder(ViewGroup viewGroup, int viewType) {
+        return new ViewHolder(getDefaultView());
+    }
+
+    @Override
+    public void onBindViewHolder(ViewHolder viewHolder, final int position) {
+        if (position < objects.size()) {
+            final T object = objects.get(position);
+
+            if (textKey == null) {
+                viewHolder.textView.setText(object.getObjectId());
+            } else if (object.get(textKey) != null) {
+                viewHolder.textView.setText(object.get(textKey).toString());
+            } else {
+                viewHolder.textView.setText(null);
+            }
+
+            if (imageKey != null) {
+                if (viewHolder.imageView == null) {
+                    throw new IllegalStateException(
+                            "Your object views must have a ParseImageView whose id attribute is 'android.R.id.icon' if an imageKey is specified");
+                }
+                if (!imageViewSet.containsKey(viewHolder.imageView)) {
+                    imageViewSet.put(viewHolder.imageView, null);
+                }
+                viewHolder.imageView.setPlaceholder(placeholder);
+                viewHolder.imageView.setParseFile((ParseFile) object.get(imageKey));
+                viewHolder.imageView.loadInBackground();
+            }
+            viewHolder.textView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    if (onClickListener != null) {
+                        onClickListener.onClick(object, position);
+                    }
+                }
+            });
```

Comment:
If we're allowing subclassing of `ViewHolder`, it might be worthwhile to move this logic into our implementation of `ViewHolder` through something like `ViewHolder.bind(ParseObject object)` and make the `ViewHolder` properties private.